watertap/core/zero_order_unit.py

```diff
@@ -0,0 +1,264 @@
+
+# Some more inforation about this module
+__author__ = "Andrew Lee"
+
+
+@declare_process_block_class('SITOBase')
+class SITOBaseData(UnitModelBlockData):
+    """
+    Standard single inlet-two outlet Unit Model Class
+    """
+
+    CONFIG = ConfigBlock()
+    CONFIG.declare('dynamic', ConfigValue(
+        domain=In([False]),
+        default=False,
+        description='Dynamic model flag - must be False',
+        doc='''All zero-order models are steady-state only'''))
+    CONFIG.declare('has_holdup', ConfigValue(
+        default=False,
+        domain=In([False]),
+        description='Holdup construction flag - must be False',
+        doc='''Zero order models do not include holdup'''))
+    CONFIG.declare('property_package', ConfigValue(
+        default=useDefault,
+        domain=is_physical_parameter_block,
+        description='Property package to use for control volume',
+        doc='''Property parameter object used to define property  calculations,
+        **default** - useDefault.
+        **Valid values:** {
+        **useDefault** - use default package from parent model or flowsheet,
+        **PhysicalParameterObject** - a PhysicalParameterBlock object.}'''))
+    CONFIG.declare('property_package_args', ConfigBlock(
+        implicit=True,
+        description='Arguments to use for constructing property packages',
+        doc='''A ConfigBlock with arguments to be passed to a property block(s)
+        and used when constructing these, **default** - None.
+        **Valid values:** {see property package for documentation.}'''))
+
+    def build(self):
+        super().build()
+
+        # Get units metadata
+        units_meta = \
+            self.config.property_package.get_metadata().get_derived_units
+
+        # Create state blocks for inlet and outlets
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["defined_state"] = True
+
+        self.properties_in = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties at inlet",
+            default=tmp_dict)
+
+        tmp_dict_2 = dict(**tmp_dict)
+        tmp_dict_2["defined_state"] = False
+
+        self.properties_out = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties at outlet",
+            default=tmp_dict_2)
+        self.properties_waste = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties at waste outlet",
+            default=tmp_dict_2)
+
+        # Create Ports
+        self.add_port("inlet", self.properties_in, doc="Inlet port")
+        self.add_port("outlet", self.properties_out, doc="Outlet port")
+        self.add_port("waste", self.properties_waste, doc="Waste port")
+
+        # Add performance variables
+        self.water_recovery = Var(
```

Comment:
In the NF and RO unit models, we've used `recovery_vol_phase` for volumetric recovery (the typical form) and `recovery_mass_phase_comp` for mass recovery of each component. In this case, `recovery_vol_phase` is the equivalent. Perhaps we want to keep this as is at the moment for the WT3 migration to prevent potential issues with renaming across files that make use of `water_recovery`, but just wanted to make note of that so that we are all aware.

Reply:
For now, I went with the WT3 naming convention (for the most part), but we should think about standardizing naming.

Reply:
I changes these to `recovery_vol` and `removal_mass_solute` to better fit the naming convention. Seeing as these models only support 1 phase, I did not include the phase index.